frontends/p4/sideEffects.cpp

```diff
@@ -323,6 +323,59 @@ bool DoSimplifyExpressions::containsHeaderType(const IR::Type* type) {
     }
     return false;
 }
+const IR::Parameter* DoSimplifyExpressions::findAlias(const IR::Parameter* param1,
+                                                      const IR::Expression* expr1,
+                                                      const IR::MethodCallExpression* mce1) {
+    auto mi = MethodInstance::resolve(mce1, refMap, typeMap);
+    const IR::Parameter* returnParam = nullptr;
+
+    for (auto param2 : *mi->substitution.getParametersInArgumentOrder()) {
+        auto arg2 = mi->substitution.lookup(param2);
+
+        auto expr2 = arg2->expression;
+        if (auto mce2 = expr2->to<IR::MethodCallExpression>()) {
+            returnParam =  findAlias(param1, expr1, mce2);
+            continue;
+        }
+        if (param1->direction == IR::Direction::In && param2->direction == IR::Direction::In) {
+            continue;
+        }
+        if (param1->type == param2->type && mayAlias(expr1, expr2)) {
+            returnParam = param2;
+            break;
+        }
+    }
+    return returnParam;
+}
+
+cstring DoSimplifyExpressions::declareTemporary(const IR::Parameter* p,
```

Comment:
The name of this function seems misleading, since it does not just declare the temporary, but it also initializes it.